test/foundry/helper/Helper.sol

```diff
@@ -0,0 +1,8 @@
+pragma solidity 0.7.6;
+import "forge-std/Test.sol";
+
+contract Helper is Test {
+    function _expect_event_emitted_from(address from) internal {
+        vm.expectEmit(false, false, false, true, address(from));
```

Comment:
in some cases, it will be out of our expectations if only check data instead of the indexed part